### PR TITLE
chore: add /api/v1 alias for create experiment endpoint [DET-4162]

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

create an alias route for create experiment under new api route

## Test Plan

- call POST /api/v1/experiments
- call GET /api/v1/experiments
- call GET /api/v1/experiments/1

## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: master/internal/core.go [hunk 1]
```diff
@@ -488,6 +488,8 @@ func (m *Master) Run() error {
 	m.echo.Static("/swagger-ui", filepath.Join(m.config.Root, "static/swagger-ui"))
 	m.echo.Static("/api/v1/api.swagger.json",
 		filepath.Join(m.config.Root, "swagger/determined/api/v1/api.swagger.json"))
+	// Support the old experiment creation endpoint under the new API route.
+	m.echo.POST("/api/v1/experiments", api.Route(m.postExperiment), authFuncs...)
```
Comment (stoksc) [inline master/internal/core.go:492]: What's going to happen for the swagger docs for this?

**Review by stoksc — APPROVED** [review]
lgtm 👍

File: master/internal/core.go (hunk 1 above)
Comment (stoksc) [reply master/internal/core.go:492]: I think we talked about this, I just forgot.

Comment (hamidzr) [reply master/internal/core.go:492]: we have https://determinedai.atlassian.net/browse/DET-4052 for that. we'd need to figure out what would be the best way of adding that. one way would be to add it through proto/scripts/swagger.py

**Comment by hamidzr** [comment]
this needs some more work. WIP.

File: common/determined_common/api/experiment.py [hunk 2]
```diff
@@ -195,7 +195,7 @@ def create_experiment(
     if additional_body_fields:
         body.update(additional_body_fields)
 
-    r = req.post(master_url, "experiments", body=body)
+    r = req.post(master_url, "api/v1/experiments", body=body)
```
Comment (hamidzr) [inline common/determined_common/api/experiment.py:198]: remove

**Comment by stoksc** [comment]
@hamidzr thoughts on just going a route like https://github.com/stoksc/determined/tree/worse/better-grpc-bypass instead?

**Comment by hamidzr** [comment]
> @hamidzr thoughts on just going a route like https://github.com/stoksc/determined/tree/worse/better-grpc-bypass instead?

hmm, thanks for taking the time to try out another approach. I did go a similar route at the beginning which pushes a "native/v0" echo handler into our gRPC proxy whereas the second approach exposes the gRPC proxy so that from our "native" echo API handers we can pass requests into the gRPC proxy. The second (https://github.com/determined-ai/determined/commit/7a99fcf2a60b0e0713749d006d9c3ad1967fc24c) approach makes a bit more sense to me but I see that going the other route results in a smaller change and potentially easier to revert when/if we have native fileupload support through grpc gateway. :thinking:

**Comment by hamidzr** [comment]
@stoksc as you suggested explicitly calling out other routes that were unintentionally captured by https://github.com/determined-ai/determined/pull/1321/files#diff-57f7323782e58c38545753e3a2839e6eR464 did the trick so I went with that and check to see we can still get /experiments/xyz and it wouldn't affect routes nested under /experiments.

File: master/internal/grpc/api.go [hunk 3]
```diff
@@ -73,15 +78,16 @@ func RegisterHTTPProxy(e *echo.Echo, port int, enableCORS bool, cert *tls.Certif
...
 	err := proto.RegisterDeterminedHandlerFromEndpoint(context.Background(), mux, addr, opts)
 	if err != nil {
 		return err
 	}
-	e.Any("/api/v1/*", func(c echo.Context) error {
-		request := c.Request()
-		if origin := request.Header.Get("Origin"); enableCORS && origin != "" {
-			c.Response().Header().Set("Access-Control-Allow-Origin", origin)
+	handler := func(c echo.Context) error {
+		if enableCORS {
```
Comment (stoksc) [inline master/internal/grpc/api.go:87]: can the cors stuff get pulled out into a separate pr?

**Review by stoksc — APPROVED** [review]
glad we got this working slightly better 👍

File: master/internal/grpc/api.go [hunk 4]
```diff
@@ -52,9 +53,8 @@ func NewGRPCServer(db *db.PgDB, srv proto.DeterminedServer) *grpc.Server {
 	return grpcS
 }
 
-// RegisterHTTPProxy registers grpc-gateway with the master echo server.
-func RegisterHTTPProxy(e *echo.Echo, port int, enableCORS bool, cert *tls.Certificate) error {
-	addr := fmt.Sprintf(":%d", port)
+// newGRPCMux creates a new gRPC server mux.
+func newGRPCMux() *runtime.ServeMux {
```
[the remark this answers is not in the document]
Comment (hamidzr) [reply master/internal/grpc/api.go:57]: yep, it's not it did look nicer to me though. makes registerHTTPProxy a bit simpler IMO

[the remark this answers is not in the document]
Comment (stoksc) [reply master/internal/grpc/api.go:56]: can we call it `newGRPCGatewayMux` then?
